Additional MSTest runner info (#38833)
dotnet/docs: docs/core/testing/unit-testing-mstest-runner-extensions.md

# MSTest runner extensions

> The MSTest runner supports extensions that can be used to customize the behavior of the test runner. These extensions are not part of the default bundle to offer biggest flexibility.

Reviewer: ```suggestion
The MSTest runner can be customized through extensions. These extension are either built-in or can be installed as NuGet packages. Extensions installed throught NuGet packages will auto-register the extensions they are holding to become available in test execution (unless you [disable it](#extension-registration)).
```

Not sure what you meant here  by "the default bundle", e.g. trx is imho already part of MSTest dependencies, because mstest -> mstest.adapter -> vstestbridge -> Microsoft.Testing.Platform.Extensions.TestReports.TrxReport 

Nevertheless I think it is valuable to document all the "major" extensions we made available, no matter if they ship together with mstest or not. Especially if they have their counterpart in vstest.

The extension registration entry might describe how extensions are registered, and what you need to do when you disable program.cs autoregistration. Each extension might also show example of the registration code? Or is that too much detail for now?

Reviewer 2: > Not sure what you meant here by "the default bundle", e.g. trx is imho already part of MSTest dependencies, because mstest -> mstest.adapter -> vstestbridge -> Microsoft.Testing.Platform.Extensions.TestReports.TrxReport

this dep is the object model, so actually as-is is not enabling the trx support

Reviewer: Can it be renamed Microsoft.Testing.Platform.Extensions.TestReports.TrxReport**Primitives**  or Abstractions? or something like that, because the current name is strongly suggesting (to me) that this is the actual extension (there is extension in the name, and no *abstract* name)

Reviewer: ```suggestion
The MSTest runner can be customized through extensions. These extension are either built-in or can be installed as NuGet packages. Extensions installed throught NuGet packages will auto-register the extensions they are holding to become available in test execution.
```

> The MSTest runner can be customized through extensions. These extension are either built-in or can be installed as NuGet packages. Extensions installed throught NuGet packages will auto-register the extensions they are holding to become available in test execution.

Reviewer: ```suggestion
The MSTest runner can be customized through extensions. These extension are either built-in or can be installed as NuGet packages. Extensions installed through NuGet packages will auto-register the extensions they are holding to become available in test execution.
```

## Test reports

A test report is a file that contains information about the execution and outcome of the tests.

### Visual Studio test reports

> The Visual Studio test result file (or TRX) is the default format for publishing test results. This extension is shipped as part of `Microsoft.Testing.Platform.Extensions` package.

Author: ```suggestion
The Visual Studio test result file (or TRX) is the default format for publishing test results. This extension is shipped as part of [Microsoft.Testing.Platform.Extensions](https://nuget.org/packages/Microsoft.Testing.Platform.Extensions) package.
```

This feature is used to understand extensions that would be registering the same command line option or the changes in available options between multiple versions of an extension (or the platform).

### Diagnostic logs

> The platform produces diagnostic logs that are helpful to understand what is happening during the execution of your test application. The following options are available to configure the produced diagnostic logs:

Reviewer: This is built-in in platform directly right? Point that out please. I was looking for a nuget to install it through.

Reviewer 2: Maybe we could have built-in section before the packages ones

Author: Or maybe instead of listing here I should move to the platfrom option on the intro page and in here only link the other section. WDYT?

Reviewer 2: what you mean by "intro page"? The Overview?

Reviewer: I think this is the correct page. Splitting it by how things are shipped does not seem like the right call. When I go here I can learn about a lot of the options that are there, and if they are already built in, then even better. Imho we just need to call out that there is no package to install.

> [!NOTE]
> Environment variables take precedence over the command line arguments.

### Hang dump files

> This extension allows you to create a dump file after a given timeout. This extension is shipped as part of `Microsoft.Testing.Platform.Extensions.HangDump` package.

Author: ```suggestion
This extension allows you to create a dump file after a given timeout. This extension is shipped as part of [Microsoft.Testing.Platform.Extensions.HangDump](https://nuget.org/packages/Microsoft.Testing.Platform.Extensions.HangDump) package.
```

| `--hangdump-timeout` | Specifies the timeout after which the dump is generated. The timeout value is specified in one of the following formats:<br/>`1.5h`, `1.5hour`, `1.5hours`<br/>`90m`, `90min`, `90minute`, `90minutes`<br/>`5400s`, `5400sec`, `5400second`, `5400seconds`. Defaults to `30m` (30 minutes). |
| `--hangdump-type` | Specifies the type of the dump. Valid values are `Mini`, `Heap`, `Triage`, `Full`. Defaults as `Full`. For more information, see [Types of mini dumps](../diagnostics/collect-dumps-crash.md#types-of-mini-dumps).|

### Crash dump files

> This extension allows you to create a crash dump file if the process crashes. This extension is shipped as part of `Microsoft.Testing.Platform.Extensions` package.

Author: We are in the process of releasing the preview to NuGet.

> This extension allows you to create a crash dump file if the process crashes. This extension is shipped as part of `Microsoft.Testing.Platform.Extensions` package.

Reviewer 3: When will that be ready?

> This extension allows you to create a crash dump file if the process crashes. This extension is shipped as part of `Microsoft.Testing.Platform.Extensions` package.

Author: Hopefully today (we are running the last manual tests), worst case tomorrow. We found some last minute issues when doing the last large manual testing round.

> This extension allows you to create a crash dump file if the process crashes. This extension is shipped as part of `Microsoft.Testing.Platform.Extensions` package.

Author: Fine to wait. I'll ping in the main chat of this PR once they are live.

> This extension allows you to create a crash dump file if the process crashes. This extension is shipped as part of `Microsoft.Testing.Platform.Extensions` package.

Author: We are live!
